diff --git a/ChangeLog b/ChangeLog
index f72b097..ddbb98f 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,7 @@
+2010-08-17  Akira TAGOH  <tagoh@redhat.com>
+
+	* src/main.c (main): make a dialog resizable.
+
 2010-06-21  Akira TAGOH  <tagoh@redhat.com>
 
 	* configure.ac: bump to 1.3.1.
diff --git a/src/main.c b/src/main.c
index e520f03..a96748a 100644
--- a/src/main.c
+++ b/src/main.c
@@ -142,7 +142,7 @@ main(int    argc,
 
 	window = gtk_dialog_new();
 	gtk_window_set_title(GTK_WINDOW (window), _("IM Chooser - Input Method configuration tool"));
-	gtk_window_set_resizable(GTK_WINDOW (window), FALSE);
+//	gtk_window_set_resizable(GTK_WINDOW (window), FALSE);
 	iconfile = g_build_filename(ICONDIR, "im-chooser.png", NULL);
 	if (!g_file_test(iconfile, G_FILE_TEST_EXISTS)) {
 		g_free(iconfile);
